### PR TITLE
Add initial support for the v2 of the API (Still WIP)

### PR DESCRIPTION
**Pull request by Redblueflame**

### CODE REVIEW

**Review by wafflecoffee — CHANGES_REQUESTED** [review]
In addition, reports cannot be created for mods:
image
Reports for versions redirect to the wrong page:
image
And there is also no visual confirmation that you have indeed submitted a report.

**Comment by venashial** [comment]
Should I create a PR for this branch to merge [`venashial:dynamic-routes`](/venashial/knossos/tree/dynamic-routes) into [`apiv2-support`](/modrinth/knossos/tree/apiv2-support)? (There shouldn't be any merge conflicts.)

File: pages/project/_id.vue [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+            v-else-if="
+              (project.server_side === 'optional' ||
+                project.server_side === 'required') &&
+              (project.client_side === 'optional' ||
+                project.client_side === 'unsupported')
+            "
+            class="side-descriptor"
+          >
+            <InfoIcon />
+            Client mod
```
Comment (wafflecoffee) [inline pages/project/_id.vue:48]:
```suggestion
            Server mod
```

**Comment by venashial** [comment]
Despite 59f685c updating this PR with the master branch, some changes weren't merged. (Likely due to project pages and components being copied/moved.) For example, now both #292 and  #252 have conflicting changes to the wording on the project editing page. A manually file comparison between the master branch and this PR could help find these discrepancies.

**Comment by Prospector** [comment]
Updated all the wording from that PR in my PR

**Comment by venashial** [comment]
59f685c did less merging than I thought. It appears that the only two commits from master to get merged into this PR since May 29 were:
- be3ed7d3806b18fc200b8249720673bd99de6f2c
- 9b4d58cfe8531ecafeee463b14c45e6e65284e87

All the other (15) commits from master need to be merged into this branch.

**Pull request merged**